config/pmd.xml

```diff
@@ -270,7 +263,7 @@
            SuppressFilterElement is a single constructor and can't be split easily -->
       <property name="violationSuppressXPath" value="//ClassOrInterfaceDeclaration[@Image='Main'
         or @Image='PackageObjectFactory' or @Image='RequireThisCheck'
-        or @Image='VariableDeclarationUsageDistanceCheck'
+        or @Image='RedundantThisCheck' or @Image='VariableDeclarationUsageDistanceCheck'
```

Comment:
it looks strange that newly added check goes to exclusions

Reply:
Due to their similarities, `RedundantThisCheck` suffers from the same `NcssCount` issue as `RequireThisCheck` does.
Specifically their `collectDeclarations`  is a big switch that author of `RequireThisCheck` has qualified as *being too hard to remove*.
(See https://github.com/checkstyle/checkstyle/blob/b0b9012d3dffbc8dd861e39c6994a8ff6a7842fe/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java#L476)

Reply:
But if there is already a suppression comment, why it is added in xpath suppressions?

Reply:
Suppression comment is for checkstyle to allow. 
Adding xpath suppression *in* `pmd.xml` is for PMD to allow.
That's what I think, and it is also what I experienced:
- Before writing comment I had checkstyle's failures
- After writing comment and before editing `pmd.xml` file I had PMD failures.